composer/utils/sftp_object_store.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Utility for uploading to and downloading from cloud object stores."""
+import io
+import os
+import sys
+import uuid
+import warnings
+from typing import Iterator, Optional, Union
+
+from paramiko import SFTPAttributes, SSHClient
+
+from composer.utils.iter_helpers import IteratorFileStream
+
+__all__ = ["SFTPObjectStore"]
+
+
+class SFTPObjectStore:
+    """Utility for uploading to and downloading to a server via SFTP."""
+
+    def __init__(self, host: str, port: int = 22, username: Optional[str] = None, key_file_path: Optional[str] = None):
+        self.ssh_client = SSHClient()
+
+        # add AutoAddPolicy
+        if key_file_path is None:
+            self.ssh_client.load_system_host_keys()
+        else:
+            try:
+                self.ssh_client.load_system_host_keys(key_file_path)
+            except IOError:
+                warnings.warn(
+                    "Host keys could not be read from {key_file_path}. Attempting to read keys from known hosts.")
+                self.ssh_client.load_system_host_keys()
+
+        # implement some retries here
+        self.ssh_client.connect(host, port, username)
+
+        self.sftp_client = self.ssh_client.open_sftp()
+
+    def close_client(self):
```

Comment:
`.close` is the standard method for cleanup

```suggestion
    def close(self):
```